src/board.py

```diff
@@ -21,3 +21,9 @@ def stringify_spaces(self):
     
     def full(self):
         return not Board.space in self.spaces
+
+    def update(self, index, token):
+        self.spaces[index] = token
+
+    def turn_count(self):
+        return 9 - self.spaces.count(Board.space)
```

Comment:
You'll want to capture this 9 value as a constant. 1) the reader will better understand the intent 2) you want to protect from situations where the business rules might change (IE 4x4 board instead of 3x3).

Reply:
Do I want it as a constant then? Maybe an instance variable is more suitable that way it can be changed from instance to instance rather than a constant that we wouldnt be able to change in the future...

Reply:
What? If it changes in the future, just change the value of the constant. A constant is called a constant so the value can't be changed at runtime.

Reply:
I'm considering a case where the user is choosing the size of his board, an instance variable would allow that to be dynamic, where a constant wouldnt. 

Maybe I'm just thinking ahead of myself though

Reply:
I would agree. That is thinking ahead given it isn't a current business requirement. The change from constant to dynamic int in the future would be painless. The protection you get from it being a constant is added clarity to future readers of the code.

Reply:
Great, ill make the adjustment